### DISCUSSION TITLE
How to migrate from MWS to SP API using a developer without Professional Amazon selling account

### DISCUSSION THREAD

**Discussion by FZ10000** [amzn/selling-partner-api-models#3776] [General]
I am an Amazon seller in need of migrating my current custom MWS application to SP API.  This would be classified as a private application.  I have used independent developers on a per project basis for this in the past.

However, my understanding is that the SP API requires the developer to have a Professional Amazon selling account and my previous independent developers do NOT have Professional Amazon selling accounts.

My questions:

1)  Is it possible for a developer to update the application using my SP API credentials?  Or is this a) not possible b) very impractical c) pose security risks?

2) If the developer is located outside the USA is it even possible for them to obtain a Professional Amazon selling account?

I find it strange that Amazon does not seemingly offer an avenue for individual sellers to have private SP API development done for them, unless they code themselves.   Maintaining a Professional Amazon selling account requires credit card information, bank account, INFORM Consumers Act information, $39.99 monthly fee, etc...  I don't think most independent developers would do that.

How would you proceed?

**Comment by stefnats** [comment]
Ah i first mistake the term "developer" in regards to Amazon universe. 
Usually "developer" means "Solution Provider" e.g. your app.
But you mean "developer" as "IT Guy who works as developer for your company" (internal or external)

In that case: It's risky and not privacy-friendly to let external developers interact with your or others' seller accounts directly via seller central.
You should set up contracts who prohibid mis-use, and even then it might be risky.

To be totally safe, you should contact developer support on advice how to get it right.

I hope i understood you correctly and answered at least some of your questions.

**Comment by FZ10000** [comment]
Yes, thank you.  

Yes I have already contacted developer support asking the same and they simply confirmed that you must have a Professional Amazon seller account in order to have API credentials.  So they confirmed that, but did not really address my scenario.

For me, there is no "IT department" or "internal IT guy", it's just me, and I have relied upon independent contractors to write custom C# code for me in the past accessing MWS.

Now I am not sure how to do that.  If I let a developer simply use my credentials to migrate my application, is that enough?  Or does SP API require someone to have full access to a seller account?

**Comment by FZ10000** [comment]
I'm bumping this up, still trying to get some help:

I need to hire a developer in order to migrate my exisiting custom MWS desktop application to SP-API.  I am the only user and I am not a developer.  I can understand the API calls, but I am a bit confused by the new "procedural" components of SP-API.   

My developer does NOT have his own Amazon API account.  Does he therefore need to actually access my Amazon account in order to update the code? Or if I provide him with my credentials will he be able to do it?

I would really appreciate any guidance here, thank you.

**Comment by SylvieManx** [comment]
Individual sellers can also register for SPAPI (MWS upgraded version). You only need an Amazon store, and then register SPAPI developer information. I have a lot of experience in this area. I migrated from MWS to SPAPI.
This is my blog: https://zhuanlan.zhihu.com/p/366253978. You can view it translated into English. Or add me on WeChat to communicate, I hope I can help you.
